src/test/java/org/mockitoutil/SafeJUnitRule.java

```diff
@@ -0,0 +1,111 @@
+package org.mockitoutil;
+
+import org.assertj.core.api.AbstractThrowableAssert;
+import org.assertj.core.api.Assertions;
+import org.junit.rules.MethodRule;
+import org.junit.runners.model.FrameworkMethod;
+import org.junit.runners.model.Statement;
+
+/**
+ * Junit rule for testing exception handling other JUnit rules, like Mockito JUnit rules.
+ * Makes it easy to assert on expected exceptions triggered by the rule under test.
+ *
+ * TODO update other tests that validate rule error handling so that they use this class.
+ */
+public class SafeJUnitRule implements MethodRule {
+
+    private final MethodRule testedRule;
+    private FailureAssert failureAssert = null;
+    private Runnable successAssert = new Runnable() { public void run() { } };
+
+    /**
+     * Wraps rule under test with exception handling so that it is easy to assert on exceptions fired from the tested rule.
+     */
+    public SafeJUnitRule(MethodRule testedRule) {
+        this.testedRule = testedRule;
+    }
+
+    public Statement apply(final Statement base, final FrameworkMethod method, final Object target) {
+        return new Statement() {
+            public void evaluate() throws Throwable {
+                try {
+                    testedRule.apply(base, method, target).evaluate();
+                    successAssert.run();
+                } catch (Throwable t) {
```

Comment:
I think `ExpectedException` can make this code a lot more readable for other developers. You can probably also take advantage of AssertJ with `expectException`